contracts/contracts/RequestManager.sol

```diff
@@ -192,16 +189,13 @@ contract RequestManager is Ownable, LpWhitelist {
 
     /// Check whether a given request ID is valid.
     modifier validRequestId(uint256 requestId) {
-        require(
-            requestId <= requestCounter && requestId > 0,
-            "requestId not valid"
-        );
+        require(requestId <= counter && requestId > 0, "requestId not valid");
```

Comment:
This is not a sufficient check for the validity of a request ID. Since we use the same counter for both claims and requests, we can have request with ID 2 and claim with ID 3. If we now try to do something with request ID 3, the check will return `true` and we will proceed, which should not happen.

I remember we talked about using one of the guaranteed nonzero fields in the `Request` struct (e.g. `validUntil` or `sender`) to check whether a given request ID is valid.

Reply:
uh. Something went wrong here. I remember changing this to checking the sender only.

Reply:
Okay this is werid. I see the changes in the changed files. Did you maybe review on a different commit?

Reply:
> Okay this is werid. I see the changes in the changed files. Did you maybe review on a different commit?

I encountered the change on the first commit, I think.